src/components/AddApplication/AuthTypeSetter.js

```diff
@@ -0,0 +1,63 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector, shallowEqual } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { useSource } from '../../hooks/useSource';
+import get from 'lodash/get';
+
+export const checkAuthTypeMemo = () => {
+    let previousAuthType;
+
+    return (newAuthType) => {
+        if (previousAuthType === newAuthType) {
+            return false;
+        }
+
+        previousAuthType = newAuthType;
+        return true;
+    };
+};
+
+export const innerSetter = ({ sourceTypes, source, appTypes, formOptions, checkAuthType, authenticationValues }) => {
+    const sourceType = sourceTypes.find(({ id }) => id === source.source_type_id);
+
+    const formValues = formOptions.getState();
+
+    const application = appTypes.find(
+        ({ id }) => id === get(formValues, 'values.application.application_type_id', undefined)
+    );
+    const supported_auth_type = application ? application.supported_authentication_types[sourceType.name][0] : '';
```

Comment:
what if `application.supported_authentication_types[sourceType.name][0] ` does not exist?

Reply:
Fixed